packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -120,6 +170,57 @@ export default class SaytDriverPlugin implements Plugin {
     return [searchTerms];
   }
 
+  /**
+   * Extracts query and products from the given response.
+   *
+   * @param response An object containing the original query and product records.
+   * @returns An object containing the query and an array of valid simplified products.
+   */
+  searchCallback(response: Results): ProductsResponseSection {
+    let filter;
+    const { query, records } = response;
+    const mappedRecords = records.map(record => {
+      try {
+        filter = this.parseRecord(record);
+      } catch(error) {
+        return;
+      }
+      const { data, firstVariant, nonvisualVariants } = filter;
+
+      return {
+        title: data.title,
+        price: nonvisualVariants[0].originalPrice,
+        imageSrc: firstVariant.productImage,
+        imageAlt: data.title,
+        productUrl: firstVariant.productImage,
+        // @TODO Handle variants
+      }
+    }).filter(Boolean);
+    return {
+      query,
+      products: mappedRecords,
+    };
+  }
+
+  /**
+   * Parses a given product record for valid data. Throws an error if data is invalid.
+   *
+   * @param record An object containing the product record data.
+   * @returns An object containing relevant product data.
+   */
+  parseRecord(record: Record): any {
+    const data = record.allMeta;
+    const firstVariant = data.visualVariants[0];
```

Comment:
What is the standard convention that is put on record structures now? The apparel demo uses the naming convention you used here, Signet, BJs, and AB use `visualVariant` and `nonvisualVariant`, CVS uses `variants` and `subVariant`, and Zoro uses just `variants` to name a few. 

Is the assumption that a client would be able to override this and/or a product transformer that would be called at some point in the search response flow in order to fit their own data structure? 

I know this has been mentioned as needing to discuss with ETL and this doesn't need to be addressed right now, but flagging it here to raise attention to it.

Reply:
@geoffcoutts Clients will need to be able to pass some sort of transformation that we can act on to provide the relevant product data. Or, the data needs to be in a standard format that we can use across all clients. The latter seems unlikely, but it's worth pushing for some sort of basic standardization.

The current solution is hardcoded for the `apparel` data. We'll need a story - possibly even more than one - to produce a solution for this. We can also refer to how Storefront dealt with this issue.